.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,75 @@
+name: Build image and push to ECR
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      app:
+        required: false
+        type: string
+        default: eip-ero-print-api
+      ecr_repo:
+        required: false
+        type: string
+        default: print-api-repo
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: corretto
+          java-version: 17
+          cache: gradle
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.DEV_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.DEV_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
+        id: build-image
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: ${{ inputs.ecr_repo }}
+          IMAGE_TAG: ${{ inputs.version }}
+        run: |
+          ./gradlew bootBuildImage
+          IMAGE=$ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
+          docker tag ${{ inputs.app }}:latest $IMAGE
+          docker push $IMAGE
+          echo "::set-output name=image::$IMAGE"
+
+      - name: Configure AWS credentials (live env)
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.LIVE_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.LIVE_AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-west-2
+
+      - name: Login to Amazon ECR (live env)
+        id: login-ecr-live
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR (live env)
```

Comment:
Doesn't look like it actually builds it - more Promote?

Reply:
Fair